lib/src/game/base_game.dart

```diff
@@ -32,8 +34,35 @@ class BaseGame extends Game with FPSCounter {
   /// Components to be removed on the next update
   final Set<Component> _removeLater = {};
 
-  /// Camera position; every non-HUD component is translated so that the camera position is the top-left corner of the screen.
-  Vector2 camera = Vector2.zero();
+  /// The camera translates the coordinate space after the viewport is applied.
+  final Camera camera = Camera();
+
+  /// The viewport transforms the coordinate space depending on your chosen
+  /// implementation.
+  /// The default implementation no-ops, but you can use this to have a fixed
+  /// screen ratio for example.
+  Viewport viewport = DefaultViewport();
+
+  /// This is overwritten to consider the viewport transformation.
+  ///
+  /// Which means that this is the logical size of the game screen area as
+  /// exposed to the canvas after viewport transformations.
+  /// This does not match the Flutter widget size; for that see [rawSize].
+  @override
+  Vector2 get size {
+    assertHasLayout();
+    return viewport.getEffectiveSize();
+  }
+
+  /// This is the original Flutter widget size, without any transformation.
```

Comment:
Not sure about this name, what do you think about canvasSize?

Reply:
sure